app/scripts/controllers/ens/index.js

```diff
@@ -0,0 +1,75 @@
+const ethUtil = require('ethereumjs-util')
+const ObservableStore = require('obs-store')
+const punycode = require('punycode')
+const Ens = require('./ens')
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+const ZERO_X_ERROR_ADDRESS = '0x'
+
+class EnsController {
+  constructor ({ ens, provider, networkStore } = {}) {
+    const initState = {
+      ensResolutionsByAddress: {},
+    }
+
+    this._ens = ens
+    if (!this._ens) {
+      const network = networkStore.getState()
+      if (Ens.getNetworkEnsSupport(network)) {
+        this._ens = new Ens({
+          network,
+          provider,
+        })
+      }
+
+      networkStore.subscribe((network) => {
```

Comment:
Should this be inside this if block?

Reply:
No? Where do you imagine this would be?

Reply:
Outside the if block? Currently this is only executed if `ens` isn't passed in to the constructor.

Reply:
Oh! Yes, good catch!